src/devices/honeywell_cm921.c

```diff
@@ -370,50 +486,104 @@ static int honeywell_cm921_decode(r_device *decoder, bitbuffer_t *bitbuffer)
         fi -= 8;
     }
     if (!seen_aa || bitrow_get_byte(bytes.bb[row], fi) != 0x35)
+    {
+        decoder_logf(decoder, 1, __func__, "did not find 35 55+");
         return DECODE_FAIL_SANITY;
+    }
+#endif
+    unsigned first_byte = start33+24;
+    uint8_t seperator_pattern[2] = { 0xff, 0x00 };
+    sep = bitbuffer_search(&bytes, row, first_byte, seperator_pattern, sizeof(seperator_pattern)*8);
+    // if no seperator, this will be the end of the buffer
+
+    if ( sep < bytes.bits_per_row[row] )
+        decoder_logf(decoder, 1, __func__, "seperator at %d", sep);
+
+    // Find Footer 0x35 (0x55*)
+    unsigned fi = sep - 8;
+    sep = sep + 16; // advance over seperator
+
+    int seen_aa = 0;
+    while (fi>8 && bitrow_get_byte(bytes.bb[row], fi) == 0x55) {
+        seen_aa = 1;
+        fi -= 8;
+    }
+    if (!seen_aa || bitrow_get_byte(bytes.bb[row], fi) != 0x35)
+    {
+        decoder_logf(decoder, 1, __func__, "did not find 35 55+");
+        //return DECODE_FAIL_SANITY;
+        continue;
+    }
+
 
-    unsigned first_byte = 24;
-    unsigned end_byte   = fi;
-    unsigned num_bits   = end_byte - first_byte;
+    //unsigned end_byte   = fi;
+    //unsigned num_bits   = end_byte - first_byte;
     //unsigned num_bytes = num_bits/8 / 2;
 
     bitbuffer_t packet = {0};
-    unsigned fpos = bitbuffer_manchester_decode(&bytes, row, first_byte, &packet, num_bits);
-    unsigned man_errors = num_bits - (fpos - first_byte - 2);
+    unsigned fpos = bitbuffer_manchester_decode(&bytes, row, first_byte, &packet, fi-first_byte); // decode as much as we can
 
-#ifndef _DEBUG
-    if (man_errors != 0)
-        return DECODE_FAIL_SANITY;
-#endif
+    if ( fpos < fi )
+    {
+        decoder_logf(decoder, 1, __func__, "manchester_fail start=%d sep=%d fail_at=%d", first_byte, sep, fpos);
+
+        char *bitrow_asprint_bits(uint8_t const *bitrow, unsigned bit_len);
```

Comment:
If we're going to promote this to a usable function, we should talk about that and it should be in a header.